### DISCUSSION TITLE
Improve Nullable<T> default checking

### DISCUSSION THREAD

**Discussion by ahdung** [dotnet/csharplang#4369] [Language Ideas]
I hope have a graceful way to check `Nullable<T>` is `null` or `default(T)`.
```c#
// now
x.SubList == null || x.SubList.Count == 0
(x.SubList?.Count ?? 0) == 0
x.SubList?.Count is null or 0

// i hope
x.SubList?.Count is default // or some else like dualdefault | none | empty
```

**Comment by huoyaoyuan** [comment]
The current blocker is that you can't check if any struct is default in a general way.

**Reply by ahdung** [reply]
I don't understand what your mean, `default(T)` is.

**Reply by khm1600** [reply]
`default(T)` is not. It only gives the default value of `T`.

**Reply by ahdung** [reply]
So, why can't `struct1 == default(Struct)` ?

**Reply by khm1600** [reply]
Structs don't have `operator==` defined by default.

**Reply by ahdung** [reply]
Interesting, this is different from my memory, wait me take a try

**Reply by ahdung** [reply]
you're right, i mistook (i almost never use custom structure).
But i don't think this is a insurmountable blocker, the complier can prevent `struct`s which is unimplemented use `==`, it also can prevent they use `is default`. partially usable still makes sense.

**Reply by yaakov-h** [reply]
`.Equals` is implemented by default.

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRjzwDcBDMAAgDMB7WygXkqgFMB3SgMXsoG9KAIQpNKmGELIAvUVAgAbeZQC+AbmL4CATgAUAEgBEdBgEsAzpTitqZBaH416AOgCiARwhl5ZnVZt2dHloASmCVA2D1XDwzYEgAY2Buejw+PEoMylQAZkoTKCThKgEAc1ZgVUozcsrlTPTMnKyCTEkZUpqqzuU8ZSA=

https://docs.microsoft.com/en-us/dotnet/api/system.valuetype.equals?view=net-5.0#remarks

**Reply by canton7** [reply]
It also boxes and uses reflection (in most cases). That might or might not matter.
